src/lib.rs

```diff
@@ -387,6 +388,17 @@ impl Server {
             }
         }
     }
+
+    /// Same as `recv()` but doesn't block longer than timeout
+    pub fn try_recv_timeout(&self, timeout: Duration) -> IoResult<Option<Request>> {
+        loop {
```

Comment:
What's the point of the `loop` if all branches of the `match` end up `return`ing?